Fix GH-15752: `finfo_file()` - parameter validation error messages

DanielEScherzer opened the pull request:
Show the correct parameter names and numbers - the errors are caused by the
second parameter being empty or having null bytes, not the first.

DanielEScherzer commented:
Note that the merge to `master` will be complicated by #15489 - error function and output are different now

devnexen commented:
> Note that the merge to `master` will be complicated by #15489 - error function and output are different now

Differences between master and release branches is a common occurence, nothing new ! lgtm once you fix the existing tests failures.

DanielEScherzer commented:
> > Note that the merge to `master` will be complicated by #15489 - error function and output are different now
> 
> Differences between master and release branches is a common occurence, nothing new ! lgtm once you fix the existing tests failures.

Done - turns out there is an existing test that was just expecting the wrong thing, so I fixed that and dropped my new test

devnexen reviewed (approved):
LCTM
